### DISCUSSION TITLE
Output of TOPAS parameters

### DISCUSSION THREAD

**Discussion by Yuri-HubL** [e0404/matRad#736] [Q&A] [answered]
Hi matRad Expert ,

Does the latest version of matRad-dev_varRBErobOpt_2.0 no longer support the output of TOPAS parameters?
I notice that under the path **matRad-dev_varRBErobOpt_2.0\topas** , there is no longer a folder called MCrun.

Did you give up using this function because the output parameters are not accurate?

your sincerely,
Yuri

**Comment by wahln** [comment]
We don't have a branch dev_varRBErobOpt_2.0? 
In general, the changes we do on this development branches are fluid - there have been some changes in using more descriptive folder names to create a topas simulation setup (e.g. with the scheme modality_date). Accuracy of the output parameter files is not a reason for that, and the mechanism for writing the parameter files stayed largely the same.

**Reply by Yuri-HubL** [reply]
Hi Wahlm，Thank you for your reply.

I did on the latest version of the matRad_example12_simpleParticleMonteCarlo still see code 
**pln.propMC.externalCalculation = true;**

But when I run it, I don't see the corresponding output file being saved. Does this mean that, now can't matrad beam and die body information, use on the TOPAS simulated

**Reply by wahln** [reply]
The reason for this is probably that we dropped the "propMC" in pln and everything that has been in propMC should now be in propDoseCalc. 
You can try running with `pln.propDoseCalc.externalCalculation = true;` instead, and I will remove the old code fragment in the example you mentioned.

Also notice that the dev_varRBErobOpt still exists, but that all our developments there were merged into dev, which means that the dev branch will see consecutive updates in the future (and dev will become the next release).

**Reply by Yuri-HubL** [reply]
hi Wahlm,I hope everything goes well with your dev version.

I replaced the old code with the new code as you said, and created a new folder MCrun under the topas folder. But there is still no output TOPAS saved.

How can I quickly find the location of propDoseCalc, externalCalculation, or any of the other functions? If there is a shortcut can help me to quickly locate?

**Reply by wahln** [reply, marked as answer]
All dose calculation algorithms in matRad are now implemented in separate classes. The TOPAS Engine is, for example, here:
https://github.com/e0404/matRad/blob/a5b8762da006ff78702f7d4065bb0bf7c110047d/doseCalc/%2BDoseEngines/matRad_TopasMCEngine.m

Everything that is a defined property in this class can be set via pln.propDoseCalc and will then be used for dose calculation. The following sequence of parameters
```matlab
pln.propDoseCalc.engine = 'TOPAS';
pln.propDoseCalc.externalCalculation = true;
pln.propDoseCalc.numHistoriesDirect = 1e6;
resultGUI_MC = matRad_calcDoseDirect(ct,stf,pln,cst,resultGUI.w);
```
Will write the TOPAS parameter files into the MCrun folder. There will be an error thrown afterwards (because there are no outputs ready), but the files will exist in the folder to be run with TOPAS.

**Reply by Yuri-HubL** [reply]
Hi Wahlm，According to your instructions, operation program error.I will attach my running code.

**pln.propDoseCalc.engine = 'MCsquare';
pln.propDoseCalc.engine = 'TOPAS';**
Once when I replaced the instruction from MCsquare to TOPAS ,the program will be a problem.
![image](https://github.com/e0404/matRad/assets/72288469/c3d7c558-66b8-4455-b3a9-88e7b01b4877)

What causes this? Can you give me some help and advice
[proton.txt](https://github.com/user-attachments/files/15912990/proton.txt)

**Reply by wahln** [reply]
As I said in my previous answer - if you set externalCalculation to true, the dose calculation will throw an error after the externalCalculation. The reason for this is that the dose calculation will not produce results, as you requested for the calculation to be done externally (and matRad will not have a result). Your parameter files should be available for running in TOPAS anyways.

**Reply by Yuri-HubL** [reply]
Thank you for your reply, I seem to see what you mean.

I also want to ask more questions, here's the output after the TOPAS parameters have been optimized, has been achieved in the phantom has a more uniform RBE weighted dose?
